app/models/govbox/api_connection_with_obo_support.rb

```diff
@@ -17,17 +17,15 @@ class Govbox::ApiConnectionWithOboSupport < ::ApiConnection
   def box_obo(box)
     raise "OBO not allowed!" if obo.present?
 
-    box.settings_obo
+    box.settings_obo.presence
   end
 
   def destroy_with_box?
     false
   end
 
   def validate_box(box)
-    box.errors.add(:obo, :not_allowed) if obo.present?
-    if tenant
-      box.errors.add(:settings_obo, :blank) if box.settings_obo.blank?
-    end
+    box.errors.add(:settings_obo, :not_allowed) if obo.present?
+    box.errors.add(:settings_obo, :invalid) if boxes.where.not(id: box.id).map(&:settings_obo).include?(box.settings_obo)
```

Comment:
Takyto pattern "load dat z db a potom kontrola na strane app je bad practice" skusme vyrobit select, ktory zisti, ze tam uz taky box je. Zaroven toto asi zlyha pri editaci, myslim, ze tu by skor mal byt unique constraint.

Reply:
Ano, [constraint](https://github.com/slovensko-digital/govbox-pro/blob/6eaba47fb1f460be9a2641399c096a361eb707a5/db/migrate/20240131112405_allow_only_unique_combinations_of_api_connection_id_and_box_settings_obo.rb) mame, len sme to chceli odchytit aj pri validacii, aby sme vedeli pridat error. Chceme teda toto nahradit nejakym selectom rozumnym? @jsuchal

Reply:
za mna by tam mala byt proste podmienka, ze nemame este taky box s tym settings_obo - toto sa da cez json query v pg spravit lahko + not na id ktore uz mas.

Reply:
Upravila som, pozri prosim.